addOns/reports/src/test/java/org/zaproxy/addon/reports/ExtensionReportsUnitTest.java

```diff
@@ -790,6 +790,50 @@ void checkAlert(JSONObject alert) {
                 instances.getJSONObject(0).getString("evidence"), is(equalTo("Test <p>Evidence")));
     }
 
+    void checkAlertJSONPlus(JSONObject alert) {
```

Comment:
This should not duplicate everything, extract a common method and have only the specific `otherinfo` checked in each method.

Reply:
Still I think we should have consistent handling of the other info in all reports (I didn't check all but the other JSON is not correct).

Reply:
yes, correct. I have only changed it for traditional-json-plus. I confirmed about that from Simon.

Reply:
ok I will make a common method for that

Reply:
Simon did suggest starting with this one :wink:

@thc202 wouldn’t this be a breaking change for the actual traditional reports? (Not that it shouldn’t be done to the other non-trads, just want to land in the right place).

Reply:
Is the confirmation the comment in the issue or was that a direct message?

Reply:
yes in the issue.

Reply:
Only if we break it, we can keep adding the otherinfo field as we do now.

Reply:
https://github.com/zaproxy/zaproxy/issues/7260#issuecomment-1406385828 is this one? That seems to be a suggestion to get to know the code. Anyway, we should not fix the issue if we are going to merge just these changes IMO.

Reply:
fwiw this is a breaking change for this report.

Reply:
I think you are correct. Please tell me what should I do now ? Should I change all the reports ?

Reply:
@psiinon can you please finalize what should be done ?

Reply:
Up to you to decide if you want to change them all (though first we all should agree that's what should be done).

Re the report being changed now, we should not break it, the other info in the general details should be brought back. The tests should also have at least two alert instances to verify that the other info is being added correctly.

Reply:
I checked all the reports and found that only these reports are using "otherinfo" and only these will be needed to be updated: 

1. Traditional JSON report 
2. Traditional JSON report with requests and responses
3. Traditional XML report 
4. Traditional JSON report with requests and responses

Tell me if any other report also need to be changed.

Tell me when you all agree to make changes in these reports, then I will start the work 😄.

Reply:
That's bad, all reports should include the other info. That list is not complete though `SARIF` and `Risk and Confidence HTML` include other info as well.

Reply:
I cannot find where `SARIF` and `Risk and Confidence HTML` are using other info. Can you help me find it please ? Here I have included the snapshots of both the reports : 

![Screenshot from 2023-01-29 15-45-24](https://user-images.githubusercontent.com/87857234/215319651-55ce926f-b467-4111-b89d-06d0659e181c.png)
![Screenshot from 2023-01-29 15-46-38](https://user-images.githubusercontent.com/87857234/215319655-d2805bda-837d-4fe5-8eee-2e4cab9fb856.png)

Reply:
@thc202 there are no tests written for xml-plus report should I write that?

Reply:
Ideally we should have tests for everything. Again, it's up to you to decide that (I don't want to push work).

Reply:
okk I will write that.

Reply:
This was not yet addressed: https://github.com/zaproxy/zap-extensions/pull/4381#discussion_r1089915714

Reply:
you mean the two alert instances part ?

Reply:
Thank you both, sorry I missed it.